share/wake/lib/system/remote_cache_runner.wake

```diff
@@ -0,0 +1,210 @@
+    def virtual job stdout stderr status runtime cputime membytes ibytes obytes = prim "job_virtual"
+    def badlaunch job error = prim "job_fail_launch"
+
+    def doit job runnerInput = match runnerInput
+        Fail e ->
+            def _ = badlaunch job e
+
+            Fail e
+        Pass input ->
+            require Pass hashKey = hashFn runnerInput
+
+            # -------------------------------------
+            # --- Search the cache for the job  ---
+            # -------------------------------------
+
+            # Search the cache for a match
+            require Pass response =
+                rscApi
+                | rscApiFindMatchingJob (mkSearchRequest input hashKey)
+            else
+                # Leave a breadcrumb and fallback to the base runner if the cache fails to respond
+                def _ = breadcrumb "Failed to search for job in the cache"
+
+                # This job isn't getting cached. That's probably preferable since the server
+                # request failed but good to keep in mind.
+                baseDoIt job (Pass input)
+
+            # If a match was found use it
+            require NoMatch = response
+            else
+                require (Match details) = response
+                else unreachable "two-constructor tuple must have one value"
+
+                def _ = breadcrumb "Found a match in the cache"
+
+                def (
+                    CacheSearchResponseMatch
+                    outputSymlinks
+                    outputDirs
+                    outputFiles
+                    stdoutBlob
+                    stderrBlob
+                    status
+                    runtime
+                    cputime
+                    mem
+                    ibytes
+                    obytes
+                ) = details
+
+                def doDownload ((CacheSearchOutputFile path mode blob): CacheSearchOutputFile) =
+                    rscApiGetFileBlob blob "{input.getRunnerInputDirectory}/{path}" mode
+
+                def blobDownloads =
+                    outputFiles
+                    | map doDownload
+
+                def stdoutDownload = rscApiGetStringBlob stdoutBlob
+                def stderrDownload = rscApiGetStringBlob stderrBlob
+
+                require Pass stdout = stdoutDownload
+                else failWithError "rsc: Failed to download stdout"
+
+                require Pass stderr = stderrDownload
+                else failWithError "rsc: Failed to download stderr"
+
+                # We don't actually care about the result here but we do need to ensure that all
+                # downloads have completed and succeeded before we continue on.
+                require Pass _ =
+                    blobDownloads
+                    | findFail
+                else failWithError "rsc: Failed to download a blob"
```

Comment:
I think all these `Failed to download` messages will probably need more context to be useful